src/hardware/mixer.cpp

```diff
@@ -96,6 +99,21 @@ static struct {
 
 Bit8u MixTemp[MIXER_BUFSIZE];
 
+float GetFadeScalar(uint8_t step)
+{
+	static constexpr std::array<float, 51> cos_scalars = {
+	        1.0f,    0.9990f, 0.9961f, 0.9911f, 0.9843f, 0.9755f, 0.9649f,
+	        0.9524f, 0.9382f, 0.9222f, 0.9045f, 0.8853f, 0.8645f, 0.8423f,
+	        0.8187f, 0.7939f, 0.7679f, 0.7409f, 0.7129f, 0.6841f, 0.6545f,
+	        0.6243f, 0.5937f, 0.5627f, 0.5314f, 0.5000f, 0.4686f, 0.4373f,
+	        0.4063f, 0.3757f, 0.3455f, 0.3159f, 0.2871f, 0.2591f, 0.2321f,
+	        0.2061f, 0.1813f, 0.1577f, 0.1355f, 0.1147f, 0.0955f, 0.0778f,
+	        0.0618f, 0.0476f, 0.0351f, 0.0245f, 0.0157f, 0.0089f, 0.0039f,
+	        0.0010f, 0.0000f};
+	constexpr uint8_t last = cos_scalars.size() - 1;
+	return cos_scalars[last - (std::min)(last, step)];
```

Comment:
I still needed parens to keep Visual Studio happy, otherwise it errored out with: `mixer.cpp(114,33): error C2589: '(': illegal token on right side of '::'`

Reply:
This happened because in includes, `"dosbox.h"` (which pulls-in `"config.h"`) is placed after `<windows.h>`.

To fix this: place `#include "mixer.h"` (which is the main include for this file) at the very top, where it belongs and (optionally) remove `#include "dosbox.h"` (as it's already pulled in via main include for this file).

Reply:
Thank you!